### PR TITLE
Furtastic image support.

### PR DESCRIPTION
**Pull request by Xenoyia**
This PR is to enable image upload support on the Furtastic platform, for accounts that have an API key enabled.

### CODE REVIEW

File: commons/src/interfaces/websites/furtastic/furtastic.file.options.interface.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { DefaultFileOptions } from '../../submission/default-options.interface';
+
+// tslint:disable-next-line: class-name
+export interface furtasticFileOptions extends DefaultFileOptions {
```
Comment (mvdicarlo) [inline commons/src/interfaces/websites/furtastic/furtastic.file.options.interface.ts:4]: nit: Please use PascalCase classes.
e6 is an exception not a rule.

File: commons/src/websites/furtastic/furtastic.file.options.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Expose } from 'class-transformer';
+import { IsArray, IsOptional, IsString } from 'class-validator';
+import { DefaultFileOptions } from '../../interfaces/submission/default-options.interface';
+import { furtasticFileOptions } from '../../interfaces/websites/furtastic/furtastic.file.options.interface';
+import { DefaultValue } from '../../models/decorators/default-value.decorator';
+import { DefaultFileOptionsEntity } from '../../models/default-file-options.entity';
+
+// tslint:disable-next-line: class-name
+export class furtasticFileOptionsEntity extends DefaultFileOptionsEntity implements furtasticFileOptions {
```
Comment (mvdicarlo) [inline commons/src/websites/furtastic/furtastic.file.options.ts:9]: nit: PascalCase

File: commons/src/websites/furtastic/furtastic.options.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { DefaultOptionsEntity } from '../../models/default-options.entity';
+import { furtasticFileOptionsEntity } from './furtastic.file.options';
+
+// tslint:disable-next-line: class-name
+export class furtastic {
```
Comment (mvdicarlo) [inline commons/src/websites/furtastic/furtastic.options.ts:5]: nit: PascalCase

File: electron-app/src/server/websites/furtastic/furtastic.service.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+export class Furtastic extends Website {
+  readonly BASE_URL: string = 'https://api.furtastic.art';
+  readonly acceptsFiles: string[] = ['jpeg', 'jpg', 'png', 'gif', 'webm'];
+  readonly acceptsSourceUrls: boolean = true;
+  readonly defaultDescriptionParser = PlaintextParser.parse;
+  readonly enableAdvertisement: boolean = false;
+
+  readonly usernameShortcuts = [
+    {
+      key: 'e6',
```
Comment (mvdicarlo) [inline electron-app/src/server/websites/furtastic/furtastic.service.ts:39]: issue: key collision

File: electron-app/src/server/websites/furtastic/furtastic.service.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    }
+    return status;
+  }
+
+  getScalingOptions(file: FileRecord): ScalingOptions {
+    return { maxSize: FileSize.MBtoBytes(100) };
+  }
+
+  preparseDescription(text: string) {
+    return UsernameParser.replaceText(text, 'e6', '@$1').replace(
```
Comment (mvdicarlo) [inline electron-app/src/server/websites/furtastic/furtastic.service.ts:59]: issue: dupes e621 code.

**Comment by mvdicarlo** [comment]
Needs some fixup.
How much is exact duplicate of e621 ? Since it looks like that is what you used as the basis of your commit.

**Comment by Xenoyia** [comment]
> Needs some fixup. How much is exact duplicate of e621 ? Since it looks like that is what you used as the basis of your commit.

New changes should fix all of the issues you found, I've been unable to test it as getting a working dev build on my environment has proved troublesome, lots of unsupported engines ^^;

**Comment by leaftail1880** [comment]
> > Needs some fixup. How much is exact duplicate of e621 ? Since it looks like that is what you used as the basis of your commit.
> 
> New changes should fix all of the issues you found, I've been unable to test it as getting a working dev build on my environment has proved troublesome, lots of unsupported engines ^^;

To test things locally use develop branch instead of master. Some new commits fixed that, also how to run this in readme is updated on develop branch.

**Comment by leaftail1880** [comment]
Also note that in develop branch there is one eslint config for all of that and `npm run codestyle` to find/fix style issues

**Comment by Xenoyia** [comment]
It seems to be working on a dev environment now!

**Comment by Xenoyia** [comment]
It seems to all work now, including multi-image submissions.

image

**Pull request closed without merging**